### PR TITLE
[USM] go TLS cleanup debug messages

### PR DESCRIPTION
**Pull request by nplanel**
### What does this PR do?

* check during 10ms /proc/pid exist on Exec event
* removing log.Debug that flood the sysprobe logs

### Motivation

cleanup and review

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/protocols/http/ebpf_gotls.go [hunk 1]
```diff
@@ -265,9 +265,21 @@ func (p *GoTLSProgram) Stop() {
 
 func (p *GoTLSProgram) handleProcessStart(pid pid) {
 	exePath := filepath.Join(p.procRoot, strconv.FormatUint(uint64(pid), 10), "exe")
-	binPath, err := os.Readlink(exePath)
+
+	var binPath string
+	var err error
+	// We receive the Exec event, /proc could be slow to update
+	end := time.Now().Add(10 * time.Millisecond)
+	for end.After(time.Now()) {
```
Comment (p-lambert) [inline pkg/network/protocols/http/ebpf_gotls.go:273]: I think it is unlikely, but its possible that this `for` loop may never execute and you'll get to a state where  `err` is `nil` and `binPath` is empty

Comment (nplanel) [reply pkg/network/protocols/http/ebpf_gotls.go:273]: yeah true, thanks in really heavy load may the kernel (and gosched) will not schedule us
let's go on the safe side

**Review by p-lambert — APPROVED** [review]

File: pkg/network/protocols/http/ebpf_gotls.go [hunk 2]
```diff
@@ -265,9 +265,22 @@ func (p *GoTLSProgram) Stop() {
...
+			binPath, err = os.Readlink(exePath)
+			if err == nil {
+				break
+			}
+			time.Sleep(time.Millisecond)
+		}
+	}
+	if err != nil {
+		// we can't access to the binary path here (pid probably ended already)
+		// there are not much we can't do and we don't want to flood the logs
```
Comment (guyarb) [inline pkg/network/protocols/http/ebpf_gotls.go:283]:
```suggestion
		// there are not much we can do and we don't want to flood the logs
```

**Review by guyarb — APPROVED** [review]
small change

File: pkg/network/protocols/http/ebpf_gotls.go (hunk 2 above)
Comment (nplanel) [reply pkg/network/protocols/http/ebpf_gotls.go]: done

**Review by kkhor-datadog — APPROVED** [review]

**Pull request merged**